ref/c/segwit_addr.c

```diff
@@ -51,9 +51,14 @@ int bech32_encode(char *output, const char *hrp, const uint8_t *data, size_t dat
     uint32_t chk = 1;
     size_t i = 0;
     while (hrp[i] != 0) {
-        if (hrp[i] >= 'A' && hrp[i] <= 'Z') return 0;
-        if (!(hrp[i] >> 5)) return 0;
-        chk = bech32_polymod_step(chk) ^ (hrp[i] >> 5);
+        int ch = hrp[i];
+        if (ch < 33 || ch > 126) {
+            return 0;
+        }
+
+        if (ch >= 'A' && ch <= 'Z') return 0;
+        if (!(ch >> 5)) return 0;
```

Comment:
This test is redundant; it's implied by `(ch < 33)` above.

Reply:
Removing